Change setpoint changes to Temporary

PRoomberg opened the pull request:
I prefer a setpoint change to be temporary.
That way a setpoint change will be active until the next scheduled setpoint change.

PRoomberg committed:
Change setpoint changes to Temporary

I prefer a setpoint change to be temporary.
That way a setpoint change will be active until the next scheduled
setpoint change.

smar000 reviewed (commented):
Are you sure that mode = 1 is temporary until next scheduled event? The modes listed above in the docstring/comments were what I had picked up from the Domoticz code, if I recall correctly. I got the impression that mode = 2 wasn't valid (but could also be just that they weren't using it). I haven't tested myself.

Also as we have had it this way from the beginning, it may impact anyone else that is using the "permanent" mode. May need to think about this one.

PRoomberg committed:
Make mode change an argument

The default mode is back at 2, but can now be changed by
adding it as an optional argument.

PRoomberg commented:
I'm sure about mode = 1 being temporary. You can see it in the EvoHome controller. Mode = 1 shows a "stopwatch" symbol. Mode = 2 show a "repeating loop" symbol.

And you're right about this breaking change. That's not good, so I've added "mode" as an optional argument. If it's not given, it's back to what it always was.

smar000 committed:
Update evogateway.py

smar000 reviewed (approved):
A parameter labelled `override_mode` may be confusing as we silently ignore it if the `until_string` is defined or if the `setpoint` is zero. I have updated to simply have a true/false flag to specify if the setpoint should be permament or not (even this has similar issue, but at least it is not directly referencing the mode).

Thanks for the PR!
